cmd/telemetrygen/internal/logs/worker_test.go

```diff
@@ -75,6 +75,28 @@ func TestRateOfLogs(t *testing.T) {
 	assert.True(t, len(exp.logs) <= 20, "there should have been less than 20 logs, had %d", len(exp.logs))
 }
 
+func TestRateOfLogsLessThan1(t *testing.T) {
+	cfg := &Config{
+		Config: common.Config{
+			Rate:          0.3,
+			TotalDuration: time.Second * 3,
+			WorkerCount:   1,
+		},
+		SeverityText:   "Info",
+		SeverityNumber: 9,
+	}
+	exp := &mockExporter{}
+
+	// test
+	require.NoError(t, Run(cfg, exp, zap.NewNop()))
+
+	// verify
+	// the minimum acceptable number of logs for the rate of .3/sec for 3 seconds
+	assert.True(t, len(exp.logs) >= 1, "there should have been 1 or more logs, had %d", len(exp.logs))
+	// the maximum acceptable number of logs for the rate of .3/sec for 3 seconds
+	assert.True(t, len(exp.logs) <= 2, "there should have been less than 3 logs, had %d", len(exp.logs))
```

Comment:
I would do this differently:

- Rate: 10/s
- Duration: 500ms
- Expected: more than 1, less than 7 (to give some room for failure)

This way, you don't need to run it for 3 seconds, while still being robust against slow runners. The way it is right now, we are introducing a 9s delay to every build we run (as opposed to 1.5s from the suggestion above).

Reply:
I agree that adding 9s to the build isn't great (I wasn't too happy about that either), but the point of this is to test float values for the rate. In my specific use case, that's testing a rate <1, which I think is the only practical use case for float values anyway. Testing for output between 1-7 is a pretty wide range for the test, too.

So if the test is meant for values <1, then it will need at least 1s to get at least 1 span, am I understanding right? So would you be ok shrinking this to ~1s?

Reply:
True, can't rate be a float higher than 1? 5.5? We are still testing how many traces were generated based on a float value, so the float can be any float, no?

Reply:
It can, but in practice there isn't much measurable difference between 5 and 6 spans/second, but there is a difference when you drop below 1 span/second -> 1 span/5 seconds. That's what I mean by the only practical use for float values, to invert the rate from spans/second to seconds/span.

With a rate >1 and a duration <1s the range of possible values is so wide (like you said, [1, 7]) that I don't see much value in the test. We don't need to test that just any float value works, because the pflag library does that for us. I'm trying to test a specific application of that float value.

In that case, I'm fine with just removing these tests from here and I can add them to our downstream code since it's my use case. I don't think there is much need for a new test besides that.

Reply:
To be honest, I'm not seeing good value for this cost: rate is already there, there's a test for it as well, and you are only changing the data type.

Reply:
No problem, I agree, I removed these tests